sentry-graphql/src/main/java/io/sentry/graphql/ExceptionReporter.java

```diff
@@ -0,0 +1,146 @@
+package io.sentry.graphql;
+
+import graphql.ExecutionResult;
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters;
+import graphql.language.AstPrinter;
+import graphql.schema.DataFetchingEnvironment;
+import io.sentry.Hint;
+import io.sentry.IHub;
+import io.sentry.SentryEvent;
+import io.sentry.SentryLevel;
+import io.sentry.exception.ExceptionMechanismException;
+import io.sentry.protocol.Mechanism;
+import io.sentry.protocol.Request;
+import io.sentry.protocol.Response;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ExceptionReporter {
+  private final boolean isSpring;
+
+  public ExceptionReporter(final boolean isSpring) {
+    this.isSpring = isSpring;
+  }
+
+  private static final @NotNull String MECHANISM_TYPE = "GraphqlInstrumentation";
+
+  public void captureThrowable(
+      final @NotNull Throwable throwable,
+      final @NotNull ExceptionDetails exceptionDetails,
+      final @Nullable ExecutionResult result) {
+    final @NotNull IHub hub = exceptionDetails.getHub();
+    final Mechanism mechanism = new Mechanism();
+    mechanism.setType(MECHANISM_TYPE);
+    mechanism.setHandled(false);
+    final Throwable mechanismException =
+        new ExceptionMechanismException(mechanism, throwable, Thread.currentThread());
+    final SentryEvent event = new SentryEvent(mechanismException);
+    event.setLevel(SentryLevel.FATAL);
+
+    final Hint hint = new Hint();
+    setRequestDetailsOnEvent(hub, exceptionDetails, event);
+
+    if (result != null) {
+      @NotNull Response response = new Response();
+      Map<String, Object> responseBody = result.toSpecification();
+      response.setData(responseBody);
```

Comment:
Can you set all the information besides the `data`?

Reply:
You mean statuscode, headers and cookies?

Reply:
I can probably attach some of it for Spring in an event processor but since the response isn't finished yet it might very well be wrong / incomplete. Would you still like me to add it?

Reply:
Can you add it after the request is made? This does not need to be GraphQL-specific since it makes sense for every integration.

Reply:
> Can you add it after the request is made? 

I don't understand. Do you mean after the response is finished? We'd have to delay sending an event to Sentry, probably put the response stuff on the scope and pick it up from there once ready. Sounds brittle and complicated.

Reply:
Yes, after the request is "finished".
I don't know the internals to tell how complicated is, but we do that with transactions and spans right, we collect all the information and "capture" at the end once everything is figured out, maybe we need something similar.
It's not a must but wishable.

Reply:
Can we do this in a follow up PR?